rpmlint/checks/BinariesCheck.py

```diff
@@ -291,6 +254,46 @@ def __init__(self, config, output):
         self.pie_exec_re = re.compile(pie_exec_re)
         self.usr_lib_exception_regex = re.compile(config.configuration['UsrLibBinaryException'])
 
+        # register all check functions
+        self.check_functions = [self.check_lto_section,
+                                self.check_no_text_in_archive,
+                                self.check_executable_stack]
+
+    # For an archive, test if any .text sections is empty
+    def check_no_text_in_archive(self, path):
+        if self.is_archive:
+            for elf_file in self.readelf_parser.section_info.elf_files:
+                for section in elf_file:
+                    if section.name == '.text' and section.size == 0:
+                        self.output.add_info('E', self.pkg, 'lto-no-text-in-archive', path)
+                        return
+
+    # Check for LTO sections
+    def check_lto_section(self, path):
+        for elf_file in self.readelf_parser.section_info.elf_files:
+            for section in elf_file:
+                if '.gnu.lto_.' in section.name:
+                    self.output.add_info('E', self.pkg, 'lto-bytecode', path)
+                    return
+
+    def check_executable_stack(self, path):
+        if not self.is_archive:
+            stack_headers = [h for h in self.readelf_parser.program_header_info.headers if h.name == 'GNU_STACK']
+            if len(stack_headers) == 0:
+                self.output.add_info('E', self.pkg, 'missing-PT_GNU_STACK-section', path)
+            elif 'E' in stack_headers[0].flags:
+                self.output.add_info('W', self.pkg, 'executable-stack', path)
+
+    def run_all_checks(self, pkg, pkgfile_path, path):
+        self.is_archive = path.endswith('.a')
```

Comment:
Instead of redeclaring this wouldn't it make much more sense in those 2 occurances of self.is_archive to simply as if path.endswith('a')? Crap the pkg.py needs to be redone using pathlib :((

Reply:
I would expect more usages of `self.is_archive` in the future.